operator/validator/controller.go

```diff
@@ -1050,6 +1041,55 @@ func (c *controller) fetchAndUpdateValidatorsMetadata(logger *zap.Logger, pks []
 	return nil
 }
 
+func (c *controller) ReportValidatorStatuses(ctx context.Context) {
+	ticker := time.NewTicker(time.Second * 30)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			start := time.Now()
+			validatorsPerStatus := make(map[validatorStatus]uint32)
+
+			for _, share := range c.validatorStore.OperatorValidators(c.operatorDataStore.GetOperatorID()) {
+				if share.IsParticipating(c.beacon.GetBeaconNetwork().EstimatedCurrentEpoch()) {
+					validatorsPerStatus[statusParticipating]++
+				}
+				meta := share.BeaconMetadata
+				if meta == nil {
+					validatorsPerStatus[statusNotFound]++
+				} else if meta.IsActive() {
+					validatorsPerStatus[statusActive]++
+				} else if meta.Slashed() {
+					validatorsPerStatus[statusSlashed]++
+				} else if meta.Exiting() {
+					validatorsPerStatus[statusExiting]++
+				} else if !meta.Activated() {
+					validatorsPerStatus[statusNotActivated]++
+				} else if meta.Pending() {
+					validatorsPerStatus[statusPending]++
+				} else if meta.Index == 0 {
+					validatorsPerStatus[statusNoIndex]++
+				} else {
+					validatorsPerStatus[statusUnknown]++
+				}
+			}
+			for status, count := range validatorsPerStatus {
+				c.logger.
+					With(zap.String("status", string(status))).
+					With(zap.Uint32("count", count)).
+					With(zap.Duration("elapsed_time", time.Since(start))).
+					Info("recording validator status")
+				recordValidatorStatus(ctx, count, status)
+			}
+		case <-ctx.Done():
+			c.logger.Info("stopping to report validator statuses. Context cancelled")
```

Comment:
`stopping to report` -> `stopped reporting`

Reply:
fixed